src/golibmc.go

```diff
@@ -184,61 +329,253 @@ func New(servers []string, noreply bool, prefix string, hashFunc int, failover b
 	}
 
 	failoverInt := 0
-	if failover {
+	if client.failover {
 		failoverInt = 1
 	}
 
 	C.client_init(
-		client._imp,
+		cn._imp,
 		(**C.char)(unsafe.Pointer(&cHosts[0])),
 		(*C.uint32_t)(unsafe.Pointer(&cPorts[0])),
 		C.size_t(n),
 		(**C.char)(unsafe.Pointer(&cAliases[0])),
 		C.int(failoverInt),
 	)
 
-	client.configHashFunction(int(hashFunctionMapping[hashFunc]))
-	client.servers = servers
-	client.prefix = prefix
-	client.noreply = noreply
-	client.disableLock = disableLock
-	return
+	cn.configHashFunction(int(hashFunctionMapping[client.hashFunc]))
+	if client.retryTimeout > 0 {
+		C.client_config(cn._imp, RetryTimeout, client.retryTimeout)
+	}
+	if client.pollTimeout > 0 {
+		C.client_config(cn._imp, PollTimeout, client.pollTimeout)
+	}
+	if client.connectTimeout > 0 {
+		C.client_config(cn._imp, ConnectTimeout, client.connectTimeout)
+	}
+	return &cn, nil
 }
 
-// SimpleNew to create a memcached client with default params
-func SimpleNew(servers []string) (client *Client) {
-	noreply := false
-	prefix := ""
-	hashFunc := HashCRC32
-	failover := false
-	disableLock := false
-	return New(
-		servers, noreply, prefix, hashFunc,
-		failover, disableLock,
-	)
+func (client *Client) putConn(cn *conn, err error) error {
+	client.lk.Lock()
+	if isBadConnErr(err) || !client.putConnLocked(cn, nil) {
+		client.lk.Unlock()
+		err1 := cn.quit()
+		if err1 != nil {
+			log.Printf("Failed cn.quit: %v", err1)
+		}
+		return err
+	}
+	client.lk.Unlock()
+	return err
```

Comment:
the err returned is not used.

Reply:
Yes, putConn () is mostly called from defer. How do you think it will be better?